src/txkube/testing/integration.py

```diff
@@ -481,6 +487,74 @@ def check_configmaps(collection):
 
 
 
+class _PodTestsMixin(object):
+    @async
+    @needs(namespace=creatable_namespaces())
+    def test_pod(self, namespace):
+        """
+        ``Pod`` objects can be created and retrieved using the ``create`` and
+        ``list`` methods of ``IKubernetesClient``.
+        """
+        return self._create_list_test(
```

Comment:
It's primarily just the general problem of Deferred-returning tests being incompatible with `given`.